packages/abi-coder/src/abi-coder.ts

```diff
@@ -188,6 +194,11 @@ export default class AbiCoder {
     }
 
     const coders = nonEmptyTypes.map((type) => this.getCoder(type));
+    if (nonEmptyTypes[0] && nonEmptyTypes[0].type === 'raw untyped slice') {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      coders[0].length = bytes.length / 8;
```

Comment:
> Does the @ts-ignore mean that this line will never run?

I _think_ it only means that the TS compiler should not run any type checks on this line. It can still run once compiled. If we are sure that we are expecting an error here, maybe we should switch to an `@ts-expect-error` instead?